Thor

jermartin77 opened the pull request:
I wish this pulled in your commit commments. So I updated the components to use a picture component and even a basic component that loads the content for the callouts and the banners. I'll add some notes inline because I don't know where some of this should go. We should also probably settle on our CSS structure before we go too deep.

jermartin77 committed:
Some basic Thor commits.

jermartin77 committed:
Merge remote-tracking branch 'origin/master' into thor

* origin/master:
  Updating splitCallout link setup to work properly
  Removing log
  Creating a TextBlock slice to support the Text slice in prismic, and adding some basic formatting around that

jermartin77 committed:
Updating to a more generic project and adding some subcomponents.

Lots of general formatting

Merge branch 'master' into thor

# Conflicts:
#	components/decorator/AppHeader.vue
#	components/decorator/app-header.vue
#	components/decorator/app-masthead.vue
#	layouts/default.vue
#	plugins/globalComponents.js

jermartin77 committed:
Removing some whitespace.

jermartin77 commented on assets/styles/core/_layout.scss:
```diff
@@ -16,7 +16,7 @@ html, body {
 
 main {
-  padding-top: $header-height;
+  // padding-top: $header-height;
```
Im getting rid of the fixed header for now, so it is a little more generic.

jermartin77 commented on assets/styles/core/_theme.scss:
```diff
@@ -0,0 +1,28 @@
+.theme-light {
```
I think this theme file could also house some predefined theme colors. Like black backgrround blue background, light gray background.

jermartin77 commented on components/banners/BannerContent.vue:
```diff
@@ -0,0 +1,44 @@
+<template>
```
This file should serve for the content layer for videos, banner images and callouts.

cbaxter713 replied:
Nice, I like it

jermartin77 commented on components/banners/HeroBanner.vue:
```diff
@@ -1,22 +1,28 @@
-    </picture>
+    <banner-content
+                :title="title"
+                :content="subtitle">
```
The `content` prop right here should match. I almost thing we build some basic pure bricks pages and then some basic content pages. I'm not sure about that yet.

jermartin77 commented on components/banners/HeroBanner.vue:
```diff
@@ -27,27 +33,46 @@
 
-<style scoped lang="scss">
+<style lang="scss">
   .hero-banner {
```
it's not scoped here so the child `BannerContent` component can inherit styles. I don't know if that is right.

cbaxter713 replied:
We don't want to do this.  There may be some cases where we don't want encapsulated css on a component, but this isn't one of them.  What needs to be done here is `banner-content` needs to be extended to take alignment props that you pass into it, which control it's own alignment css from within the component.  Wherever you use `banner-content` you can style the host element for things like positioning, z-index, etc...but if you need styles of the internals, those should be handled by passing down props.

Any given component should have ZERO knowledge of any other components (i.e. classes used within, etc).  Each component is an API, with props that are passed in and events that are emitted out...thats it

jermartin77 commented on components/slices/ImageBanner.vue:
```diff
@@ -0,0 +1,74 @@
+  }
+</script>
+
+<style lang="scss">
```
Not scoped so child content can inherit styles.

cbaxter713 replied:
I think this should be scoped, and then don't style dom elements that live in the child components...but add classes to your child components and style those directly

jermartin77 commented:
Other notes. In the `ImageBanner` in prismic I used repeatable fields for links (like if you needed 2 buttons) I didn't see those come in in the data. 

Can't get the text as HTML stuff working. 

I'd like to use the headings selected in prismic and have those render in HTML.

cbaxter713 commented on components/utility/PictureComponent.vue:
```diff
@@ -0,0 +1,25 @@
+<template>
+  <picture class="picture-component">
+    <source :srcset="desktopUrl" media="(min-width: 800px)">
```
There should also be a `desktopBreakpoint` prop, which could have a default value of 800, but could be configurable to any value wherever this component is used

Commit:
Cleanup of prop typing and usage of prismic-dom helper functions to get things working

cbaxter713 reviewed (commented):
I made a commit to this branch to get things working.  Take a look, just some minor cleanup of prop types and usage of prismic-dom: https://github.com/cbaxter713/nuxt-prismic-template/pull/4/commits/a3a7d96d9a8308afe366a95bf64ca8c586c99ea2 .  

Looking pretty good man!  Let's chat about this in the morning.  I do disagree about offloading the decision of h1 vs h2 vs h3 for titles to the content creator...I think that is a bad idea.  Too much room for error for there to be multiple h1's on the page, and could veer too far from the design...but lets chat about it.

Also, I think we need to figure out the best way to have no application logic in any display components, as in any usage of prismic-dom or anything like that.  Ideally all of our slices, banner components, etc will have their props setup to take in exactly what they need for display, and the Pages that compose those components will be responsible for using prismic-dom or anything else to configure the data into the form it needs to be for those display components.  We really need to do that to get into a habit of making truly reusable components, and isolate all application logic to the state management and top level Page components only

jermartin77 committed:
Updating the prismic attributes for copy.

jermartin77 committed:
Merge remote-tracking branch 'origin/master' into thor

* origin/master:
  Removing old breakpoints
  Matching js mediaQueries to css and adding a layout-container component to replace the global container css
  Tying the base-picture into the media query strategy
  Centralizing the breakpoints object and exposing for usage where it is needed
  Adding responsive mq library and leveraging into a responsive-display component and aspect-ratio component with a breakpoint
  Renaming directory from `presentation` to `base`
  Minor clean up
  Adding in a whole bunch of core base elements

# Conflicts:
#	assets/styles/core/core.scss
#	components/slices/TextBlock.vue

jermartin77 committed:
fixing merge issue.

jermartin77 committed:
More base styling

Merge remote-tracking branch 'origin/master' into thor

* origin/master:
  Removing old breakpoints
  Matching js mediaQueries to css and adding a layout-container component to replace the global container css
  Tying the base-picture into the media query strategy
  Centralizing the breakpoints object and exposing for usage where it is needed
  Adding responsive mq library and leveraging into a responsive-display component and aspect-ratio component with a breakpoint
  Renaming directory from `presentation` to `base`
  Minor clean up
  Adding in a whole bunch of core base elements

# Conflicts:
#	assets/styles/core/core.scss
#	components/slices/TextBlock.vue

jermartin77 committed:
Coordinating theme into a component.

jermartin77 committed:
Creating a Theme component

jermartin77 committed:
More update to themes and buttons so it inherits style from the theme.

jermartin77 committed:
Removing commented out code.

cbaxter713 commented on components/utility/PictureComponent.vue:
```diff
@@ -0,0 +1,25 @@
+<template>
```
Don't need this component anymore, lets use the BasePicutre

cbaxter713 commented on components/banners/HeroBanner.vue:
```diff
@@ -1,53 +1,75 @@
+                :content="$prismic.asHtml(content)"
+                class="theme-dark">
+    </banner-content>
+    <picture-component
```
use the BasePicture component here

jermartin77 committed:
More updates.

jermartin77 committed:
More updates.
